frame/composable-maths/src/dex/price.rs

```diff
@@ -0,0 +1,62 @@
+use composable_traits::{defi::Rate, math::SafeArithmetic};
+use frame_support::ensure;
+use sp_runtime::{
+	traits::{AtLeast32Bit, Convert, Saturating},
+	ArithmeticError, DispatchError, FixedPointNumber, SaturatedConversion,
+};
+
+/// Uniswap V2 TWAP
+/// https://docs.uniswap.org/protocol/V2/concepts/core-concepts/oracles
+pub fn compute_initial_price_cumulative<Conv, Balance>(
```

Comment:
compute_initial_price_cumulative = compute_next_price_cumulative(0, 0, 1, rate)

Reply:
Nice catch, didn't really thought about it :D